src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionResult.ExpressionResult.cs

```diff
@@ -48,15 +49,7 @@ public override bool ContainingScopeHasAsyncKeyword()
                 if (scope == null)
                     return null;
 
-                // If the selection ends up mapping to the name on the right in `x.y` or `x?.y`, then grab
-                // the full expr `x.y` or `x?.y` as the part to extract.
-
-                scope = SyntaxFactory.GetStandaloneExpression(scope);
-
-                // If we're in a chain in a conditional access expression, then grab the top of that ?. chain.
-                scope = scope.GetRootConditionalAccessExpression() ?? scope;
-
-                return scope;
+                return CSharpSyntaxFacts.Instance.GetRootStandaloneExpression(scope);
```

Comment:
this much more strongly expresses what this function is really trying to do.  this will also be a useful helper for other features that need to work with expressions and will not want to work with a subportion of a CAE.